### PR TITLE
v1/api: add DeleteBlueprint endpoint

### PR DESCRIPTION
**Pull request by ezr-ondrej**
Adds Delete endpoint for Blueprint.

Refs HMS-3417

### CODE REVIEW

**Comment by ezr-ondrej** [comment]
Drafting this, so https://github.com/osbuild/image-builder/pull/998 can go in first, I'll rebase on top of that.

File: internal/v1/api.yaml [hunk 1]
```diff
@@ -483,6 +483,14 @@ paths:
...
               schema:
                 $ref: '#/components/schemas/HTTPErrorList'
+    delete:
+      summary: delete a blueprint
+      description: |
+        Deletes all versions of Blueprint, the compose will still count towards quota.
+      operationId: deleteBlueprint
+      responses:
+        204:
+          description: OK
```
Comment (avitova) [inline internal/v1/api.yaml:493]: What about 404 here? :thinking: Which leads me to the fact that we do not really test 404 in handlers, right?

Comment (avitova) [reply internal/v1/api.yaml:493]: Also, can you possibly add the following?

```
      tags:
        - blueprint
```

Comment (ezr-ondrej) [reply internal/v1/api.yaml:493]: that's true, good catch, let me add 404 and test for it here :+1:
We can revisit other handlers in follow-ups :)

**Comment by lzap** [comment]
LGTM

For the record, deletion orphans composes, which is expected and by design: https://github.com/osbuild/image-builder/blob/704d9b18ff181fcf09cf422ef18c699d25de7d87/internal/db/migrations-tern/009_create_table_blueprint_versions.sql#L18

**Review by avitova — DISMISSED** [review]
LGTM:)

**Comment by lzap** [comment]
Please rebase.

**Comment by ezr-ondrej** [comment]
Rebased :)

**Comment by ezr-ondrej** [comment]
Huh did content sources changed their API? :thinking:

**Comment by lzap** [comment]
Weird, irrelevant tests are failing.

@croissanne any ideas? maybe `main` is broken, I tried to initiate new build but it straight up failed (maybe no permissions):

https://ci.ext.devshift.net/job/osbuild-image-builder-pr-check/

**Comment by jrusz** [comment]
> Weird, irrelevant tests are failing.
> 
> @croissanne any ideas? maybe `main` is broken, I tried to initiate new build but it straight up failed (maybe no permissions):
> 
> https://ci.ext.devshift.net/job/osbuild-image-builder-pr-check/

You have to type `/retest` in a comment like I did above to re-trigger the pr-check... Also if there are issues with this particular job feel free to ping me, it runs the IQE plugin tests for which I'm responsible. I'm going to take a look what's wrong with this and fix it.

**Comment by croissanne** [comment]
```
 HTTP response body: {"errors":[{"status":500,"title":"Could not get repository list","detail":"error: code=500, title=Could not get repository list, detail=json: unknown field \"module\" \n"}]}
```

**Comment by ezr-ondrej** [comment]
/retest

Content sources apparently fixed it :) let's see

**Review by lzap — APPROVED** [review]

**Pull request merged**